src/net/metaplaylist/isobmff_patcher.ts

```diff
@@ -0,0 +1,461 @@
+            this.filterBox(boxtype, boxdata, output.length) :
+            boxdata;
+        output = concat(output, concatData);
+        pos += size;
+      }
+      return output;
+    }
+
+    /**
+     * Process specific box and its children
+     * @param {string} boxtype
+     * @param {Uint8Array} data
+     * @param {number} filePos
+     * @param {string} _path
+     */
+    filterBox(
+      boxtype: string,
+      data: Uint8Array,
+      filePos: number,
+      _path?: string
+    ): Uint8Array{
+      let output = new Uint8Array(0);
+      const path = _path ? (_path + "." + boxtype) : boxtype;
+      if (this.compositeBoxesToParse.indexOf(boxtype) >= 0) {
+        output = data.subarray(0,8);
+        let pos = 8;
+        while(pos < data.length) {
+          const {
+            size: childSize,
+            boxtype: childBoxType,
+          } = this.checkBox(data.subarray(pos, pos + 8));
+          const outputChildBox =
+            this.filterBox(
+              childBoxType,
+              data.subarray(pos, pos + childSize),
+              filePos+pos,
+              path
+            );
+          output = concat(output, outputChildBox);
+          pos += childSize;
+        }
+        if (output.length !== data.length) {
+          output = concat(itobe4(output.length), output.subarray(4, output.length));
+        }
+      } else {
+        switch(boxtype) {
+          case "styp":
+            output = this.styp(data);
+            break;
+          case "tfhd":
+            output = this.tfhd(data);
+            break;
+          case "mfhd":
+            output = this.mfhd(data);
+            break;
+          case "trun":
+            output = this.trun(data);
+            break;
+          case "sidx":
+            output = this.sidx(data, false);
+            break;
+          case "tfdt":
+            output = this.tfdt(data);
+            break;
+          case "mdat":
+            output = data;
+            break;
+          case "mvhd":
+            output = this.mvhd(data);
+            break;
+          case "tkhd":
+            output = this.tkhd(data);
+            break;
+          default:
+            output = data;
+            break;
+        }
+        // if (output.toString() !== data.toString()) {
+        //   console.log(boxtype);
+        //   console.log("");
+        // }
+      }
+      return output;
+    }
+
+    tkhd(input: Uint8Array): Uint8Array {
+      const version = input[8];
+      let output = new Uint8Array(0);
+      if (version === 1) {
+        output = concat(
+          input.subarray(0,36),
+          itobe8(0),
+          input.subarray(44, input.length)
+        );
+      } else {
+        output = concat(
+          input.subarray(0,28),
+          itobe4(0),
+          input.subarray(32, input.length)
+        );
+      }
+      return output;
+    }
+
+    mvhd(input: Uint8Array): Uint8Array {
+      const version = input[8];
+      let output = new Uint8Array(0);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this useless assignment to local variable "output". ([typescript:S1854](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS1854))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=178&issues=AWNyk0oJTmDVX2bfxyav&open=AWNyk0oJTmDVX2bfxyav)

[ISSUE_KEY]: AWNyk0oJTmDVX2bfxyav